src/test/resources/com/puppycrawl/tools/checkstyle/grammar/java15/InputAstRegressionSealedAndPermits.txt

```diff
@@ -0,0 +1,156 @@
+PACKAGE_DEF -> package [2:0]
+|--ANNOTATIONS -> ANNOTATIONS [2:47]
+|--DOT -> . [2:47]
+|   |--DOT -> . [2:39]
+|   |   |--DOT -> . [2:28]
+|   |   |   |--DOT -> . [2:22]
+|   |   |   |   |--DOT -> . [2:11]
+|   |   |   |   |   |--IDENT -> com [2:8]
+|   |   |   |   |   `--IDENT -> puppycrawl [2:12]
+|   |   |   |   `--IDENT -> tools [2:23]
+|   |   |   `--IDENT -> checkstyle [2:29]
+|   |   `--IDENT -> grammar [2:40]
+|   `--IDENT -> java15 [2:48]
+`--SEMI -> ; [2:54]
+CLASS_DEF -> CLASS_DEF [4:0]
+|--MODIFIERS -> MODIFIERS [4:0]
+|   |--LITERAL_PUBLIC -> public [4:0]
+|   `--LITERAL_SEALED -> sealed [4:7]
+|--LITERAL_CLASS -> class [4:14]
+|--IDENT -> InputAstRegressionSealedAndPermits [4:20]
+|--LITERAL_PERMITS -> permits [4:55]
+|   |--IDENT -> Circle [4:63]
+|   |--COMMA -> , [4:69]
+|   |--IDENT -> Square [4:71]
+|   |--COMMA -> , [4:77]
+|   `--IDENT -> Rectangle [4:79]
+`--OBJBLOCK -> OBJBLOCK [4:89]
+    |--LCURLY -> { [4:89]
+    `--RCURLY -> } [5:0]
+CLASS_DEF -> CLASS_DEF [7:0]
+|--MODIFIERS -> MODIFIERS [7:0]
+|   `--FINAL -> final [7:0]
+|--LITERAL_CLASS -> class [7:6]
+|--IDENT -> Circle [7:12]
+|--EXTENDS_CLAUSE -> extends [7:19]
+|   `--IDENT -> InputAstRegressionSealedAndPermits [7:27]
+|--IMPLEMENTS_CLAUSE -> implements [7:62]
+|   `--IDENT -> Squircle [7:73]
+`--OBJBLOCK -> OBJBLOCK [7:82]
+    |--LCURLY -> { [7:82]
+    `--RCURLY -> } [8:0]
+CLASS_DEF -> CLASS_DEF [10:0]
+|--MODIFIERS -> MODIFIERS [10:0]
+|   `--LITERAL_SEALED -> sealed [10:0]
+|--LITERAL_CLASS -> class [10:7]
+|--IDENT -> Rectangle [10:13]
+|--EXTENDS_CLAUSE -> extends [10:23]
+|   `--IDENT -> InputAstRegressionSealedAndPermits [10:31]
+|--IMPLEMENTS_CLAUSE -> implements [11:4]
+|   `--IDENT -> Cloneable [11:15]
+|--LITERAL_PERMITS -> permits [12:4]
```

Comment:
I would say that it should be PERMITS_CLAUSE to make it similar to EXTENDS_CLAUSE and IMPLEMENTS_CLAUSE

Reply:
Agreed. Done.